doc: Enhance rule documentation (#50)
salesforce/eslint-plugin-lwc: docs/rules/valid-api.md

> # Validate `api` decorator usage (valid-api)

Reviewer: ```suggestion
# Validate `@api` decorator usage (valid-api)
```

> The following restriction applies the `api` decorator.

Reviewer: ```suggestion
The following restrictions apply to the `@api` decorator:
```

> -   The `api` decorator can only be applied to class fields and class methods.

Reviewer: ```suggestion
-   Apply to class fields and class methods only.
```

> -   Properties and method should be unique per class.

Reviewer: ```suggestion
-   Fields and methods should be unique per class.
```

Reviewer: @pmdartus it seems like only the first rule applies to `@api`, the rest seem to be about field and method names. Am I misunderstanding? 

Also, the first rule uses "fields" so I changed the subsequent instances of "properties" to "fields," is that OK?

Author: All the rules apply to field and fields and APIs annotated with the `@api` decorator. Maybe it should be a little more explicit. For instance, the next line should be understood as _Fields and methods [anotated with `@api` decorator] can't start with `on`. The `on` prefix is reserved to bind event handlers_.

Do you think it worth adding the implicit _anotated with `@api` decorator_ on all the lines?

> -   Properties and methods can't start with `on`. The `on` prefix is reserved to bind event handlers.

Reviewer: ```suggestion
-   Fields and methods can't start with `on`. The `on` prefix is reserved to bind event handlers.
```

> -   Properties and methods can't with `data`, `slot` and `part`. The `data` prefix is reserved by LWC.

Reviewer: ```suggestion
-   Fields and methods can't start with `data`, `slot` or `part`. These names are reserved by LWC.
```

> -   Properties and methods can't be named `slot` or `part`. Those names are reserved by LWC.

Reviewer: ```suggestion
```

> -   Boolean properties should only be initialized with `false`. By initializing a public property to `true`, the consumer component can't set its value to `false` via the template.

Reviewer: ```suggestion
-   Boolean properties must be initialized with `false`. By initializing a public property to `true`, the consumer component can't set its value to `false` via the template.
```
